### PR TITLE
Fix:  Exception when parsing log #20966 

### LINKED ISSUE #20966: Exception when parsing log
### Apache Airflow version

2.1.4

### What happened

[2022-01-19 13:42:46,107] {taskinstance.py:1463} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1165, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1283, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1313, in _execute_task
    result = task_copy.execute(context=context)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py", line 367, in execute
    final_state, remote_pod, result = self.create_new_pod_for_operator(labels, launcher)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py", line 521, in create_new_pod_for_operator
    final_state, remote_pod, result = launcher.monitor_pod(pod=self.pod, get_logs=self.get_logs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/cncf/kubernetes/utils/pod_launcher.py", line 148, in monitor_pod
    timestamp, message = self.parse_log_line(line.decode('utf-8'))
UnicodeDecodeError: 'utf-8' codec can't decode bytes in position 16413-16414: invalid continuation byte

### What you expected to happen

_No response_

### How to reproduce

_No response_

### Operating System

Debian GNU/Linux 10 (buster)

### Versions of Apache Airflow Providers

apache-airflow-providers-amazon==2.2.0
apache-airflow-providers-celery==2.0.0
apache-airflow-providers-cncf-kubernetes==2.0.2
apache-airflow-providers-docker==2.1.1
apache-airflow-providers-elasticsearch==2.0.3
apache-airflow-providers-ftp==2.0.1
apache-airflow-providers-google==5.1.0
apache-airflow-providers-grpc==2.0.1
apache-airflow-providers-hashicorp==2.1.0
apache-airflow-providers-http==2.0.1
apache-airflow-providers-imap==2.0.1
apache-airflow-providers-microsoft-azure==3.1.1
apache-airflow-providers-mysql==2.1.1
apache-airflow-providers-postgres==2.2.0
apache-airflow-providers-redis==2.0.1
apache-airflow-providers-sendgrid==2.0.1
apache-airflow-providers-sftp==2.1.1
apache-airflow-providers-slack==4.0.1
apache-airflow-providers-sqlite==2.0.1
apache-airflow-providers-ssh==2.1.1

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

_No response_

### Anything else

Always on a specific docker container.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by microhuang**
When the task outputs some strange characters, decoding will lead to "UnicodeDecodeError" error due to inconsistent encoding. We can try-catch it or ignore it by specifying the parameter: errors="backslashreplace" .

e.g.

```
            line = ''
            for raw_line in iter(self.sub_process.stdout.readline, b''):
                try:
                    line = raw_line.decode(output_encoding).rstrip()
                except UnicodeDecodeError as err:
                    print(err, output_encoding, raw_line)
                self.log.info("%s", line)
```

closes: https://github.com/apache/airflow/issues/20966

### CODE REVIEW

**Review by uranusjr — APPROVED** [review]

**Comment by uranusjr** [comment]
Would it be possible to add a test for this?

**Comment by microhuang** [comment]
> Would it be possible to add a test for this?

 Okay, I'll add the test code later.

**Comment by potiuk** [comment]
Static checks need fixing - I recmmend installing pre-commit and `commit --amend` - you will get them fixed for you automatically.

**Review by potiuk — APPROVED** [review]

**Comment by eladkal** [comment]
Tests is failing:
```
airflow/providers/cncf/kubernetes/utils/pod_manager.py:202: error: Incompatible
types in assignment (expression has type "str", variable has type "bytes") 
[assignment]
                        line = line.decode('utf-8', errors="backslashrepla...
                               ^
airflow/providers/cncf/kubernetes/utils/pod_manager.py:203: error: Argument 1
to "parse_log_line" of "PodManager" has incompatible type "bytes"; expected
"str"  [arg-type]
                        timestamp, message = self.parse_log_line(line)
                                                                 ^
```

File: airflow/providers/cncf/kubernetes/utils/pod_manager.py [hunk 1]
```diff
@@ -199,7 +199,8 @@ def follow_logs(since_time: Optional[DateTime] = None) -> Optional[DateTime]:
                     ),
                 )
                 for line in logs:
-                    timestamp, message = self.parse_log_line(line.decode('utf-8'))
+                    line = line.decode('utf-8', errors="backslashreplace")
+                    timestamp, message = self.parse_log_line(line)
                     self.log.info(message)
```
Comment (uranusjr) [inline airflow/providers/cncf/kubernetes/utils/pod_manager.py:204]:
```python
                for raw_line in logs:
                    line = raw_line.decode('utf-8', errors="backslashreplace")
                    timestamp, message = self.parse_log_line(line)
```

**Comment by eladkal** [comment]
@microhuang can you please fix the test?

**Comment by eladkal** [comment]
@microhuang do you wish to finish the work on this PR?

**Comment by Siraj-Qazi** [comment]
@microhuang where you @?


> @microhuang do you wish to finish the work on this PR?

**Comment by jakubno** [comment]
Only formatting isn't right. It's almost 3 months without any activity. Is it okay if someone else takes over it?

**Comment by eladkal** [comment]
> Only formatting isn't right. It's almost 3 months without any activity. Is it okay if someone else takes over it?

If @microhuang doesn't wish to finish this PR, you can take over and open a new PR

**Comment by uranusjr** [comment]
You could also check out the branch, commit onto it, and open a new PR. The PR would belong to you, but those existing commits on the branch will belong to @microhuang.

**Pull request closed without merging**